settings/R/read.settings.R

```diff
@@ -234,67 +234,236 @@ check.settings <- function(settings) {
   scipen = getOption("scipen")
   options(scipen=12)
 
-  # check for and try to fix deprecated settings structure
-  settings <- fix.deprecated.settings(settings)
+  settings <- check.database.settings(settings)
   
-  # check database secions if exist
-  dbcon <- "NONE"
-  if (!is.null(settings$database)) {
-    # check all databases
-    for (name in names(settings$database)) {
-      if(name != "dbfiles") # 'dbfiles' is kept in <database>, but isn't actually a db settings block
-        settings$database[[name]] <- check.database(settings$database[[name]])
-    }
+  if(!is.null(settings$database$bety)) {
+    dbcon <- db.open(settings$database$bety)
+  } else {
+    dbcon <- NULL
+  }
+  
+  # make sure there are pfts defined
+  if (is.null(settings$pfts) || (length(settings$pfts) == 0)) {
+    logger.warn("No PFTS specified.")
+  }
 
-    # check bety database
-    if (!is.null(settings$database$bety)) {
-      # should runs be written to database
-      if (is.null(settings$database$bety$write)) {
-        logger.info("Writing all runs/configurations to database.")
-        settings$database$bety$write <- TRUE
-      } else {
-        settings$database$bety$write <- as.logical(settings$database$bety$write)
-        if (settings$database$bety$write) {
-          logger.debug("Writing all runs/configurations to database.")
-        } else {
-          logger.warn("Will not write runs/configurations to database.")
-        }
-      }
+  # check to make sure a host is given
+  if (is.null(settings$host$name)) {
+    logger.info("Setting localhost for execution host.")
+    settings$host$name <- "localhost"
+  }
+  
+  # check if there is either ensemble or sensitivy.analysis
+  if (is.null(settings$ensemble) && is.null(settings$sensitivity.analysis)) {
+    logger.warn("No ensemble or sensitivity analysis specified, no models will be executed!")
+  }
 
-      # check if we can connect to the database with write permissions
-      if (settings$database$bety$write && !db.exists(params=settings$database$bety, TRUE, table='users')) {
-        logger.severe("Invalid Database Settings : ", unlist(settings$database))
-      }
+  settings <- papply(settings, check.run.settings, dbcon=dbcon)
+  
+  # check meta-analysis
+  if (is.null(settings$meta.analysis) || is.null(settings$meta.analysis$iter)) {
```

Comment:
Since you merged already maybe you realized that this was just a reorganization by me, not new logic. But I agree with your suggestion to wrap it all in a `!is.null...`. I've also had plenty of trouble with MA running when I don't think it should in the past, though not recently.